packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -735,41 +733,46 @@ void NativeReanimatedModule::performOperations() {
   react_native_assert(uiManager_ != nullptr);
   const auto &shadowTreeRegistry = uiManager_->getShadowTreeRegistry();
 
-  shadowTreeRegistry.visit(surfaceId_, [&](ShadowTree const &shadowTree) {
-    shadowTree.commit(
-        [&](RootShadowNode const &oldRootShadowNode)
-            -> RootShadowNode::Unshared {
-          PropsMap propsMap;
-          for (auto &[shadowNode, props] : copiedOperationsQueue) {
-            auto family = &shadowNode->getFamily();
-            react_native_assert(family->getSurfaceId() == surfaceId_);
-            propsMap[family].emplace_back(rt, std::move(*props));
+  std::unordered_map<SurfaceId, PropsMap> propsMapBySurface;
+
+  for (auto const &[shadowNode, props] : copiedOperationsQueue) {
+    SurfaceId surfaceId = shadowNode->getSurfaceId();
+    auto family = &shadowNode->getFamily();
+    react_native_assert(family->getSurfaceId() == surfaceId);
+    propsMapBySurface[surfaceId][family].emplace_back(rt, std::move(*props));
```

Comment:
Maybe it makes sense to populate `propsMapBySurface` directly in `updateProps` function? On the other hand, I don't think having it as a separate step makes much difference in terms of execution time and performance.

Reply:
I was thinking that it is more readable (and kinda related to neighbour code) in current place (at least for me), but if you would like to put it in `updateProps`, I'm open for any changes